kbase-extension/static/kbase/js/common/cellComponents/tabs/jobStatus/jobStateList.js

```diff
@@ -193,13 +109,152 @@ define([
         });
     }
 
-    function factory() {
+    /**
+     * create a job state list instance
+     *
+     * the config should be an object with a property 'jobManager', which executes
+     * the job actions available as part of the job status table
+     *
+     * @param {object} config
+     * @returns jobStateList instance
+     */
+    function factory(config) {
         const widgetsById = {},
             bus = Runtime.make().bus(),
-            listeners = {};
+            listeners = {},
+            { jobManager } = config;
 
         let $container, tableBody;
 
+        /**
+         * Kick off a batch job action
+         *
+         * @param {event} e - event
+         *
+         * The target element's "data-" properties encode the action to be performed:
+         *
+         * - data-action - either "cancel" or "retry"
+         * - data-target - the job status of the jobs to perform the action on
+         */
+
+        function doBatchJobAction(e) {
+            const el = e.target;
+            const action = el.getAttribute('data-action'),
+                target = el.getAttribute('data-target');
+
+            // valid actions: cancel or retry
+            if (['cancel', 'retry'].includes(action)) {
+                jobManager[`${action}JobsByStatus`](target);
+            }
+        }
+
+        /**
+         * Execute an action for a single job
+         *
+         * @param {event} e - event
+         *
+         * The target element's "data-" properties encode the action to be performed:
+         *
+         * - data-action - "cancel", "retry", or "go-to-results"
+         * - data-target - the job ID of the job to perform the action on
+         */
+        function doSingleJobAction(e) {
+            const el = e.target;
+            e.stopPropagation();
+            const action = el.getAttribute('data-action'),
+                target = el.getAttribute('data-target');
+
+            if (['cancel', 'retry', 'go-to-results'].includes(action)) {
+                if (action === 'go-to-results') {
+                    // switch to results tab
+                    return jobManager.viewJobResults(target);
+                }
+                return jobManager[`${action}Job`](target);
+            }
+        }
+
+        function createActionsDropdown(events) {
```

Comment:
this function is almost entirely untouched -- the only change is adding in the call to `doBatchJobAction`